naga/src/back/spv/writer.rs

```diff
@@ -169,8 +169,14 @@ impl Writer {
             .push(Instruction::store(variable.id, container_id, None));
 
         let element_pointer_id = self.id_gen.next();
-        let element_pointer_type_id =
-            self.get_pointer_id(ir_types, element_ty, spirv::StorageClass::Function)?;
+        let element_pointer_type_id = match element_ty {
+            LookupType::Handle(handle) => {
+                self.get_pointer_id(ir_types, handle, spirv::StorageClass::Function)?
+            }
+            LookupType::Local(local_type) => {
+                self.get_local_pointer_id(local_type, spirv::StorageClass::Function)
+            }
+        };
```

Comment:
Should we pack this in one single `get_pointer_id` method?

BTW `get_pointer_id` never returns Err.

Reply:
I think we should update `get_pointer_id`.

Reply:
> address the TODO that's currently in get_local_pointer_id

This is hard as we cannot pack LookupType in pointer (due to infinite struct size), except if we box it. Another alternative is to simply put u32 for local type, but this could get pretty chaotic.

I think boxing solution is the most nice one. What do you think?

Reply:
What I hand in mind is adding a new `ValuePointer` variant. Similar to:

https://github.com/gfx-rs/wgpu/blob/43cb730d588a06ae64d91ce98afbef09d5613596/naga/src/lib.rs#L751

Reply:
LocalType has pointers encoded in LocalType::Value field.